jwst/associations/lib/rules_level3_base.py

```diff
@@ -614,7 +614,8 @@ def finalize(associations):
 format_product = FormatTemplate(
     key_formats={
         'source_id': ['s{:05d}', 's{:s}'],
-        'expspcin': ['{:02d}']
+        'expspcin': ['{:2.2s}']
```

Comment:
Never remembering how the formatting works, will this '0'-fill the field? I think the expspcin always comes with 2 characters, but it would be good to take a defensive stance and ensure that is the case if given only 1 character.

Reply:
I had to look it up:

    print("'{:0>2.2s}'".format("a"))
    '0a'
    print("'{:0>2.2s}'".format("ab"))
    'ab'

Reply:
I had to look it up as well but I'm using {:0>2} which works with strings and intergers
```
>>> "{:0>2}".format(2)
'02'
>>> "{:0>2}".format('0c')
'0c'
>>> "{:0>2}".format('c')
'0c'
```
but 
```
"{:0>2.2}".format(2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Precision not allowed in integer format specifier
```